### PR TITLE
Fix: Reflect Updated Service Notes Immediately in Asset Details without page reload

### LINKED ISSUE #9225: Edited notes are not updating in service details of Asset Details Page
#### Describe the Bug
- When editing the service detail notes, the updated notes are not reflected in the "Service Details" section immediately. The changes are only visible after the page is reloaded.

#### To Reproduce
- Steps to reproduce the behavior:

  - Open Assets tab.
  - Open any asset and edit the already existed notes of service history.
  - The edited notes are not updated in the service details notes immediately.

#### Expected Behavior
- The updated notes should appear in the "Service Details" section immediately after the changes are saved, without requiring a page reload.
Image

### PR DESCRIPTION
**Pull request by Rishith25**
## Proposed Changes

- Fixes #9225 
- Updated the AssetServiceEditModal component to include a refetch call for refreshing the service details data after editing notes.
- Ensured that the "Service Details" section dynamically reflects the updated notes without requiring a page reload.

Before
image

After
image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced asset management with a new data refresh capability.
	- Integrated refetch functionality to ensure the latest asset data is displayed after updates.

- **Improvements**
	- Improved clarity and maintainability of the `ShiftDetails` component's button properties without altering existing functionality. 

- **Bug Fixes**
	- Improved responsiveness of the UI by allowing dynamic data updates without altering the existing user interface.

### CODE REVIEW

**Review by Jacobjeevan — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

File: src/components/Assets/AssetManage.tsx [hunk 1]
```diff
@@ -595,6 +599,7 @@ const AssetManage = (props: AssetManageProps) => {
             setServiceEditData({ ...serviceEditData, open: false })
           }
           handleUpdate={() => serviceRefetch()}
+          refetch={refetch}
```
Comment (rithviknishad) [inline src/components/Assets/AssetManage.tsx]:
```suggestion
          handleUpdate={() => {
            serviceRefetch();
            refetch();
          }}
```

File: src/components/Assets/AssetServiceEditModal.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ export const AssetServiceEditModal = (props: {
   service_record: AssetService;
   handleClose: () => void;
   handleUpdate: () => void;
+  refetch: () => void;
```
Comment (rithviknishad) [inline src/components/Assets/AssetServiceEditModal.tsx]:
```suggestion
```

File: src/components/Assets/AssetServiceEditModal.tsx [hunk 3]
```diff
@@ -55,6 +56,7 @@ export const AssetServiceEditModal = (props: {
       });
       props.handleUpdate();
       props.handleClose();
+      props.refetch();
```
Comment (rithviknishad) [inline src/components/Assets/AssetServiceEditModal.tsx]:
```suggestion
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by Rishith25** [comment]
@rithviknishad Done the changes

**Pull request merged**